src/Microsoft.Tye.Hosting/ReplicaRegistry.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Tye.Hosting.Model;
+
+namespace Microsoft.Tye.Hosting
+{
+    public class ReplicaRegistry : IDisposable
+    {
+        private readonly IDictionary<ServiceType, IReplicaInstantiator> _replicaInstantiators;
+        private StateStore _store;
+
+        public ReplicaRegistry(Model.Application application, IDictionary<ServiceType, IReplicaInstantiator> replicaInstantiators)
+        {
+            _replicaInstantiators = replicaInstantiators;
+            _store = new StateStore(application);
+        }
+
+        public async Task Reset()
+        {
+            await PurgeAll();
+            _store.Reset(true, true);
+        }
+
+        public async Task WriteReplicaEvent(ReplicaEvent replicaEvent)
+        {
+            var serviceType = replicaEvent.Replica.Service.ServiceType;
+            if (_replicaInstantiators.TryGetValue(serviceType, out var instantiator))
+            {
+                var serialized = await instantiator.SerializeReplica(replicaEvent);
+                await _store.WriteEvent(new StoreEvent
+                {
+                    ServiceType = serviceType,
+                    State = replicaEvent.State,
+                    SerializedEvent = serialized
+                });
+            }
+        }
+
+        private async Task PurgeAll()
+        {
+            var events = await _store.GetEvents();
+            var tasks = new List<Task>(events.Count);
+
+            foreach (var @event in events)
+            {
+                if (_replicaInstantiators.TryGetValue(@event.ServiceType, out var instantiator))
+                {
+                    var replica = await instantiator.DeserializeReplicaEvent(@event.SerializedEvent);
+                    tasks.Add(instantiator.HandleStaleReplica(replica));
+                }
+            }
+
+            await Task.WhenAll(tasks);
```

Comment:
Should we log anything here?

Reply:
Also do we need any error handling around this? Or is the idea that error handling is inside each of the processors?

Reply:
Yeah, the processors already catch the relevant exception.
I don't know which exceptions to expect at this level since every processor has a different approach for destroying his stale replicas.